sap/adt/dataelement.py

```diff
@@ -0,0 +1,185 @@
+        """Setter for Label Heading element"""
+
+        self._metadata.data_element.label_heading = value
+
+    def normalize(self):
+        """Validate Data Element setup before save"""
+        if self._metadata.data_element.type == 'domain':
+            self._metadata.data_element.data_type = ''
+            self._metadata.data_element.data_type_length = '0'
+            self._metadata.data_element.data_type_decimals = '0'
+        if self._metadata.data_element.type == 'predefinedAbapType':
+            self._metadata.data_element.type_name = ''
+
+        # pylint: disable=line-too-long
+        self._metadata.data_element.label_short_length = self._metadata.data_element.label_short_length if not self._metadata.data_element.label_short_length else LABELS_LENGTH['short']
+        # pylint: disable=line-too-long
+        self._metadata.data_element.label_medium_length = self._metadata.data_element.label_medium_length if not self._metadata.data_element.label_medium_length else LABELS_LENGTH['medium']
+        # pylint: disable=line-too-long
+        self._metadata.data_element.label_long_length = self._metadata.data_element.label_long_length if not self._metadata.data_element.label_long_length else LABELS_LENGTH['long']
+        # pylint: disable=line-too-long
+        self._metadata.data_element.label_heading_length = self._metadata.data_element.label_heading_length if not self._metadata.data_element.label_heading_length else LABELS_LENGTH['heading']
+
+        # pylint: disable=line-too-long
+        self._metadata.data_element.deactivate_input_history = self._metadata.data_element.deactivate_input_history if self._metadata.data_element.deactivate_input_history is not None else False
+        # pylint: disable=line-too-long
+        self._metadata.data_element.change_document = self._metadata.data_element.change_document if self._metadata.data_element.change_document is not None else False
+        # pylint: disable=line-too-long
+        self._metadata.data_element.left_to_right_direction = self._metadata.data_element.left_to_right_direction if self._metadata.data_element.left_to_right_direction is not None else False
+        # pylint: disable=line-too-long
+        self._metadata.data_element.deactivate_bidi_filtering = self._metadata.data_element.deactivate_bidi_filtering if self._metadata.data_element.deactivate_bidi_filtering is not None else False
+
+    def validate(self):
+        """Validate Data Element setup"""
+
+        if self._metadata.data_element.type == 'domain' and not self._metadata.data_element.type_name:
+            raise SAPCliError('Domain name must be provided (--domain_name) if the type (--type) is "domain"')
+        if self._metadata.data_element.type == 'predefinedAbapType' and not self._metadata.data_element.data_type:
+            # pylint: disable=line-too-long
+            raise SAPCliError('Data type name must be provided (--data_type) if the type (--type) is "predefinedAbapType"')
```

Comment:
Good idea but this part knows nothing about command line parameters.

Reply:
I wanted to have it as part of DataElement object, so I did. What do you offer? How should I handle it?

Reply:
Return error codes and print the right error message in the cli part. Exceptions are not good for flow control anyways.

Reply:
ok, I probably do understand. Thanks

Reply:
Done in commit 471cfed